group/assignment6/RearrangeCars.cpp

```diff
@@ -35,36 +31,29 @@ void RearrangeCars::moveCar(vector<int> &currentStates, vector<int> &carPosition
 	carPosition[currentStates[to]] = to;
 }
 
-void RearrangeCars::generateSequenceOfMoves(const vector<int> &endStates, vector<Move> &sequenceOfMoves) {
-	vector<int> currentStates = this->startStates; // startStates will remain unchanged
-	vector<int> carPosition = findCarPositions(currentStates);
-
-	int emptySlotId = findEmptySlotId();
-	int currCarId, currCarSlotId;
-
-	while (true) {
-		currCarId = endStates[emptySlotId];
-		if (currCarId == -1)	// We reached slot which should be empty in the end
-			currCarId = findMisplacedCar(carPosition, endStates);
-		if (currCarId == -1) break;
-		currCarSlotId = carPosition[currCarId];
-		moveCar(currentStates, carPosition, currCarSlotId, emptySlotId);
-		sequenceOfMoves.push_back(Move(currCarId, make_pair(currCarSlotId, emptySlotId)));
-		emptySlotId = currCarSlotId;
-	}
-}
+bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>>* constrains) {
+	if (!constrains)	// Constrains are not set
+		return true;
 
-bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>>& constrains) {
 	if (constrains[emptySlotId].size() == 0)	// There are no constraints for this parking space, any car can be moved here
 		return true;
 
-	if ((constrains[emptySlotId]).find(carId) == (constrains[emptySlotId].end()))
+	if (((*constrains)[emptySlotId]).find(carId) == ((*constrains)[emptySlotId].end()))
 		return true;
 
 	return false;
 }
 
-int RearrangeCars::findMisplacedCarWithConstraints(vector<int> &carPosition, vector<int> &currentStates, const vector<int> &endStates, const vector<set<int>> &constraints, vector<Move> &sequenceOfMoves, int &emptySlotId, int &numOfCarsOnDisiredPosition) {
+int RearrangeCars::numberOfCarsOnDesiredPosition(const vector<int>& carPosition, const vector<int>& endStates) {

```

Comment:
placedCarsAmount, numberOfPlacedCars, I don't like current name because it should be numberOfCarsOnTheirDesiredPositions otherwise it kind of sounds that we check car numbers on one desired position. But if you don't agree leave it :)

Reply:
It makes sense, I will rename it.